src/openMVG/multiview/solver_resection_p5pfr_test.cpp

```diff
@@ -0,0 +1,630 @@
+          -1.326581000000000e+03, -7.138680000000001e+02, -1.234743000000000e+03, -6.222770000000000e+02, -1.461412100000000e+03;
+  pt3D << 3.175360000000000e+00, 2.531220000000000e+00, 2.420890000000000e+00, 7.578520000000000e-01, 3.880860000000000e+00, 
+          -4.679980000000000e-01, 1.964960000000000e-01, -1.863250000000000e+00, 1.648030000000000e-01, 4.177460000000000e-01, 
+          -8.605090000000000e-01, -1.507400000000000e+00, -1.828080000000000e+00, 1.208810000000000e+00, 2.980880000000000e-01;
+
+  // SOLUTIONS
+  std::vector<resection::M> solutions;
+  Mat R1 = Mat(3, 3);
+  R1 << -9.430073299811607e-01, 2.928814581476584e-01, -1.579798312288744e-01, 
+          -3.071682103315301e-01, -5.834954588038342e-01, 7.517850358427121e-01, 
+          1.280033834065588e-01, 7.574651813968776e-01, 6.402043680012413e-01;
+  Vec3 t1;
+  t1 << 3.351461753495843e+00, 3.662505952393360e-01, 1.522470547430807e+00;
+  double f1 = 8.754207183076484e+03;
+  Vec r1 = Vec(1);
+  r1 << -2.974934229711877e+01;
+  resection::M m1(R1, t1, r1, f1);
+  solutions.push_back(m1);
+  Mat R2 = Mat(3, 3);
+  R2 << -9.288195037630559e-01, 2.583819805318196e-01, -2.655806498331628e-01, 
+          -3.682532345104426e-01, -5.643279703727857e-01, 7.388663594504280e-01, 
+          4.103516421250264e-02, 7.840744186564097e-01, 6.193088254713202e-01;
+  Vec3 t2;
+  t2 << 3.273710373948777e+00, 3.480925794418662e-01, 1.335812819189698e+00;
+  double f2 = 6.523300619965637e+03;
+  Vec r2 = Vec(1);
+  r2 << -1.716688911283097e+01;
+  resection::M m2(R2, t2, r2, f2);
+  solutions.push_back(m2);
+  Mat R3 = Mat(3, 3);
+  R3 << 6.261772405396446e-01, 7.517419626249591e-01, -2.068479757187163e-01, 
+          7.785113680902823e-01, -6.173581864945597e-01, 1.130881042477301e-01, 
+          -4.268621773305278e-02, -2.318466976191867e-01, -9.718153094177535e-01;
+  Vec3 t3;
+  t3 << -2.286549972461777e+00, -1.360137887842060e+00, -5.183599359915888e-01;
+  double f3 = 7.215366134863186e+03;
+  Vec r3 = Vec(1);
+  r3 << -1.884020746278653e+01;
+  resection::M m3(R3, t3, r3, f3);
+  solutions.push_back(m3);
+  Mat R4 = Mat(3, 3);
+  R4 << -2.156527230855822e-01, -6.509050028038228e-01, -7.278850048948177e-01, 
+          -8.734050807538640e-01, 4.619070646021064e-01, -1.542900793440778e-01, 
+          4.366434105069141e-01, 6.024653857240441e-01, -6.681152528308791e-01;
+  Vec3 t4;
+  t4 << 5.861602062248285e-01, 5.581995729744164e-01, 8.108474186437322e-01;
+  double f4 = 1.770851635941407e+03;
+  Vec r4 = Vec(1);
+  r4 << -2.945748397114161e-01;
+  resection::M m4(R4, t4, r4, f4);
+  solutions.push_back(m4);
+  std::sort(solutions.begin(), solutions.end(), sortM);
+
+  // PROCESS
+  std::vector<resection::M> models;
+  resection::P5PfrSolver::Solve(pt2D, pt3D, 1, &models);
+  std::sort(models.begin(), models.end(), sortM);
+
+  // TEST
+  double eps = 1e-4;
+  EXPECT_TRUE(testSolutions(solutions, models, eps));
+}
+
+TEST(Resection_P5Pfr, Test01)
```

Comment:
all these TestXX are more or less the same but just with different data, or are you testing some special cases (eg coplanar points or test with only two valid solutions...)? In the latter case it's better if you rename the test accordingly. If they are just the same test but with different data keep it like that.